idaes/models/properties/modular_properties/base/generic_property.py

```diff
@@ -85,6 +86,7 @@
     _valid_VL_component_list as bub_dew_VL_comp_list,
 )
 from idaes.models.properties.modular_properties.phase_equil.henry import HenryType
+from idaes.models.properties.modular_properties.eos.ceos import Cubic
```

Comment:
I would like it to be part of the eos module. The `critical_properties` can probably be built with other thermodynamic models, but they're not in place yet. Right now, I can calculate them from the method in `ceos.py` which is the only reason why I am having to import `Cubic` into `generic_property.py`. How do I avoid it, and what default value should apply to `critical_properties`?

Reply:
I would introduce the critical properties as normal properties. Then, each EoS module would need to define the method to use.

Reply:
Does that mean it should be treated sort of the same way as `phases_in_equilibrium` or `bubble_dew_method`? What I have in mind is that the critical properties method is defined for now in `ceos.py`. It could stay there, and the property package (eg. `ASU_PR.py`, `BT_PR.py`) should specify from where the properties are built.

Reply:
Not quite, or at least not for now. For now, I would assume that the critical properties method is tied to the EoS chosen, so once you choose the EoS you have chosen the method for critical properties as well.

What I am suggesting is that we treat all critical properties like any other property (e.g., cp_mol_phase). That is , we add the critical properties to the metadata dict with a build-on-demand method which is defined in in the generic StateBlock. This method the calls out to the EoS class for the method to calculate the critical properties, e.g. `Cubic.temperature_crit`.

Did that help clarify my suggestion?

Reply:
Yes that helps, thanks Andrew

Reply:
One little hiccup: the other properties like `cp_mol_phase` are defined in `equation_of_state` which is phase-specific. But mixture critical properties are not related to a phase which is why I defined it differently compared to other properties.

Reply:
Ahh... that issue. That complicates things a bit, as we now need to be able to specify what method to use (similar to `bubble_dew_method`), but have the added complexity that that method might rely on EoS specific parameters.

So, as a first suggestion, we could have a critical properties argument which we expect to point to an EoS module - w can then do a check to make sure that EoS module is used for at least one of the phases in the system (to keep things simple for now; lets not allow a different EoS for critical properties than is used for a phase in the system).

@dallan-keylogic Any thoughts on this?

Reply:
My first thought is that the critical point should be calculated for a defined phase equilibrium pair. You can also have critical points for liquid-liquid equilibrium with lower and upper consolute points. I need to take a closer look at this PR to see if that's practical for what you're doing.

Reply:
@dallan-keylogic I am not sure that makes sense really, and either way doesn't matter until you have a VLLE system. So I wouldn't do that yet (I regret building the framework to try to support multiple liquid phases as it is).